### PR TITLE
fix: add -r to publish and set node-path to true

### PR DESCRIPTION
**Pull request by Ibrahimmaga**
# Pull Request

## 📖 Description
1. binary path require configuration
    we will use "scripts-prepend-node-path=true" in .npmrc folder and it will solve the problem
2. npm ERR! 404 Not Found - PUT https://registry.npmjs.org/@microsoft%2ffast-foundation - Not found
    The PUT is use to insert into an API. Here when publish is writing to npm, it need to have a permission. We need to provide
3. npm token with write access.
   .npmrc config communicate with npm and write to //registry.npmjs.org/:_authToken=${NPM_TOKEN}. However this will expose the token, So I moved it to the workflow and use -r as beachball publish docs suggest.

4. Update $NPM_PAT to make sure it has read and write access policies

### 🎫 Issues
#4556

### General

- [ ] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [ ] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [x] I have modified an existing component
- [x] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [x] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: .github/workflows/cd-publish.yml [hunk 1]
```diff
@@ -42,5 +42,5 @@ jobs:
    
     - name: Publish NPM changed packages
       run: |
-        yarn publish-ci -y  --access public $NPM_PAT 
+        yarn publish-ci -y -r --access public $NPM_PAT
```
Comment (EisenbergEffect) [inline .github/workflows/cd-publish.yml:45]: Just want to make sure that `$NPM_PAT` isn't a typo.

Comment (EisenbergEffect) [reply .github/workflows/cd-publish.yml:45]: Oh, probably "public access token". I thought it was a typo for "path". Ignore me.

**Review by EisenbergEffect — APPROVED** [review]

File: .github/workflows/cd-publish.yml (hunk 1 above)
Comment (Ibrahimmaga) [reply .github/workflows/cd-publish.yml:45]: sounds like a typo. it is without the $. but we need $ in CI. some people use ${NPM_PAT}, which does't work. So far $NPM_PAT is working for us.

**Review by nicholasrice — APPROVED** [review]

File: .github/workflows/cd-publish.yml (hunk 1 above)
Comment (awentzel) [reply .github/workflows/cd-publish.yml:45]: `-r` is default, so it should not be needed.

Comment (awentzel) [reply .github/workflows/cd-publish.yml:45]: I'm not seeing that NPM_PAT has been updated for read/write permissions.  Last update was October.  
Screen Shot 2021-04-12 at 10 29 57 AM

Comment (Ibrahimmaga) [reply .github/workflows/cd-publish.yml:45]: let updated

Comment (Ibrahimmaga) [reply .github/workflows/cd-publish.yml:45]: we can replace -r with "https://registry.npmjs.org/:_authToken=$NPM_PAT." we just can't put it in .npmrc for security reason.

Comment (awentzel) [reply .github/workflows/cd-publish.yml:45]: You can leave the code as is with `-r` then as I agree it shouldn't be in .npmrc.  Thanks for the clarification.

**Review by awentzel — APPROVED** [review]

**Pull request merged**